cargo-near/src/util/mod.rs

```diff
@@ -156,14 +157,38 @@ pub struct CompilationArtifact {
 pub(crate) fn compile_project(
     manifest_path: &CargoManifestPath,
     args: &[&str],
-    env: Vec<(&str, &str)>,
+    mut env: Vec<(&str, &str)>,
     artifact_extension: &str,
+    hide_warnings: bool,
 ) -> anyhow::Result<CompilationArtifact> {
+    let mut final_env = BTreeMap::new();
+
+    if hide_warnings {
+        env.push(("RUSTFLAGS", "-Awarnings"));
+    }
+
+    for (key, value) in env {
+        match key {
+            "RUSTFLAGS" => {
+                let rustflags: &mut String = final_env
+                    .entry(key)
+                    .or_insert_with(|| std::env::var(key).unwrap_or_default());
+                if !rustflags.is_empty() {
+                    rustflags.push(' ');
+                }
+                rustflags.push_str(value);
+            }
+            _ => {
+                final_env.insert(key, value.to_string());
```

Comment:
curious, why use a map instead of just a vec of `(K, V)` if you're not checking the duplication of keys on insert?

Reply:
Well, actually, for `RUSTFLAGS`, we're checking to ensure we don't replace previous entries, but instead concatenate them. So it's here to allow us quickly index the map, instead of having to iterate the whole `Vec<(K, V)>`.